File: Source/core2/E133.Business/Models/Ingredient.cs
namespace E133.Business.Models
{
    public class Ingredient 
    {
        public long Id { get; set; }
		
        public int SubrecipeId { get; set; }
		
        public string Name { get; set; }
		
        public Quantity Quantity { get; set; }
		
        public Ingredient Replacement { get; set; }
		
        public string Requirements { get; set; }
    }
}

File: Source/app/shared/models/ingredient.ts
import {Quantity, QuantityDto} from "app/shared/models/quantity";

export class Ingredient implements IngredientDto {
  id: number;
  subrecipeId: number;
  name: string;
  quantity: Quantity;
  requirements: string;
  replacement: Ingredient[];
  state: string;

  constructor();
  constructor(dto: IngredientDto);
  constructor(dto?: IngredientDto) {
    (<any>Object).assign(this, dto);

    if (dto) {
      this.quantity = new Quantity(dto.quantity);
      this.replacement = (dto.replacement ? new Ingredient(dto.replacement) : undefined);
    }

    this.state = "";
  }
}
export interface IngredientDto {
  id: number;
  subrecipeId: number;
  name: string;
  quantity: QuantityDto;
  requirements: string;
  replacement: IngredientDto[];
}
